File: 0074-search-a-2d-matrix/0074-search-a-2d-matrix.java
class Solution {
    public boolean searchMatrix(int[][] matrix, int target) {
        int s=0;
        int e=matrix.length*matrix[0].length-1;
        int r=matrix[0].length;
        while(s<=e){
            int mid=(s+e)/2;
            if(matrix[mid/r][mid%r]==target){
                return true;
            }
            if(matrix[mid/r][mid%r]>target){
                e=mid-1;
            }else{
                s=mid+1;
            }
        }
        return false;
    }
}

File: 0074-search-a-2d-matrix/0074-search-a-2d-matrix.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        l=0
        r=len(matrix)*len(matrix[0])-1
        d=len(matrix[0])
        ans=-1
        while(l<=r):
            mid=l+(r-l)//2
            if(matrix[mid//d][mid%d]==target):
                return True
            if(matrix[mid//d][mid%d]>target):
                r=mid-1
            else:
                l=mid+1
        return False
                       
            
        